app/src/main/java/com/stevesoltys/seedvault/plugins/StoragePlugin.kt

```diff
@@ -1,13 +1,17 @@
 package com.stevesoltys.seedvault.plugins
 
 import android.app.backup.RestoreSet
-import androidx.annotation.WorkerThread
-import com.stevesoltys.seedvault.settings.Storage
 import java.io.IOException
 import java.io.InputStream
 import java.io.OutputStream
 
-interface StoragePlugin {
+interface StoragePlugin<T> {
+
+    /**
+     * Returns true if the plugin is working, or false if it isn't.
+     * @throws Exception any kind of exception to provide more info on the error
+     */
+    suspend fun test(): Boolean
```

Comment:
Wonder if something like [Result4k](https://github.com/fork-handles/forkhandles/tree/trunk/result4k) would be helpful here?

Reply:
yeah not only here, but all over the codebase. so far it is mostly following the old java paradigm, that you have to catch and handle exceptions everywhere (except the compile isn't insisting on it).